File: api/store1/src/main/java/com/mrcl/store1/admin/dto/AdminOrderDetailsResponse.java
package com.mrcl.store1.admin.dto;

import com.mrcl.store1.entity.OrderStatus;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.List;

public record AdminOrderDetailsResponse(
        Long id,
        String trackingNumber,
        OrderStatus status,
        BigDecimal totalPrice,
        int totalQuantity,
        Instant dateCreated,
        Instant dateUpdated,
        String userEmail,
        AdminCustomerRow customer,
        AdminAddressRow shippingAddress,
        AdminAddressRow billingAddress,
        List<AdminOrderItemRow> items
) {
}

File: ui-frontent/loja/src/app/common/admin/AdminOrderDetailsResponse.ts
export interface AdminOrderDetailsResponse {

  id: number;
  trackingNumber: string;
  status: string;

  totalPrice: number;
  totalQuantity: number;

  dateCreated: string;
  dateUpdated: string;

  userEmail: string | null;

  customer: {
    id: number;
    firstName: string;
    lastName: string;
    email: string;
  };

  shippingAddress: {
    street: string;
    city: string;
    state: string;
    country: string;
    zipCode: string;
  };

  billingAddress: {
    street: string;
    city: string;
    state: string;
    country: string;
    zipCode: string;
  };

  items: {
    id: number;
    productId: number;
    quantity: number;
    unitPrice: number;
    imageUrl: string;
  }[];

}